core/domain/suggestion_jobs_one_off.py

```diff
@@ -0,0 +1,101 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for suggestions."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import ast
+import logging
+
+from core import jobs
+from core.domain import html_validation_service
+from core.domain import suggestion_services
+from core.platform import models
+
+(suggestion_models,) = models.Registry.import_models([models.NAMES.suggestion])
+
+
+class SuggestionMathRteAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks for existence of math components in the suggestions."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        suggestion = suggestion_services.get_suggestion_from_model(item)
+        html_string_list = suggestion.get_all_html_content_strings()
+        html_string = ''.join(html_string_list)
+        if html_validation_service.check_for_math_component_in_html(
+                html_string):
+            yield ('Suggestion with Math', item.id)
+
+    @staticmethod
+    def reduce(key, values):
+        yield (
+            '%d suggestions have Math components in them.' % len(values))
+
+
+class SuggestionMathMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """A one-time job that can be used to migrate the Math components in the
+    suggestions to the new Math Schema.
+    """
+    _ERROR_KEY = 'validation_error'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        suggestion = suggestion_services.get_suggestion_by_id(item.id)
+        try:
+            suggestion.validate()
+        except Exception as e:
+            logging.error(
+                'Suggestion %s failed validation: %s' % (item.id, e))
+            yield (
+                SuggestionMathMigrationOneOffJob._ERROR_KEY,
+                'Suggestion %s failed validation: %s' % (item.id, e))
+            return
+        suggestion.convert_html_in_suggestion_change(
+            html_validation_service.add_math_content_to_math_rte_components)
+
+        try:
+            suggestion.validate()
+        except Exception as e:
+            logging.error(
+                'Suggestion %s failed validation after migration: %s' % (
+                    item.id, e))
+            yield (
+                SuggestionMathMigrationOneOffJob._ERROR_KEY,
```

Comment:
I think you should have 2 different error keys. This will make it easier to read the output.

Reply:
done